packages/client/components/ReviewRequestToJoinOrgModal.tsx

```diff
@@ -0,0 +1,127 @@
+import styled from '@emotion/styled'
+import graphql from 'babel-plugin-relay/macro'
+import React, {useState, Suspense} from 'react'
+import {PreloadedQuery, usePreloadedQuery} from 'react-relay'
+import DialogContainer from './DialogContainer'
+import DialogContent from './DialogContent'
+import DialogTitle from './DialogTitle'
+import PrimaryButton from './PrimaryButton'
+import SecondaryButton from './SecondaryButton'
+import {ReviewRequestToJoinOrgModalQuery} from '../__generated__/ReviewRequestToJoinOrgModalQuery.graphql'
+import useQueryLoaderNow from '../hooks/useQueryLoaderNow'
+import Checkbox from './Checkbox'
+import {PALETTE} from '../styles/paletteV3'
+
+const ButtonGroup = styled('div')({
+  marginTop: '24px',
+  display: 'flex',
+  width: '100%',
+  justifyContent: 'flex-end'
+})
+
+const query = graphql`
+  query ReviewRequestToJoinOrgModalQuery {
+    viewer {
+      teams {
+        id
+        name
+        teamMembers(sortBy: "preferredName") {
+          userId
+          preferredName
+        }
+        organization {
+          name
+        }
+      }
+    }
+  }
+`
+
+interface TeamListProps {
+  queryRef: PreloadedQuery<ReviewRequestToJoinOrgModalQuery>
+  requestCreatedBy: string
+}
+
+const TeamsList = (props: TeamListProps) => {
+  const {queryRef, requestCreatedBy} = props
+  const data = usePreloadedQuery<ReviewRequestToJoinOrgModalQuery>(query, queryRef)
+  const [selectedTeams, setSelectedTeams] = useState<string[]>([])
+  const teams = data.viewer.teams
+  const sortedTeams = teams
+    .slice()
+    .sort((a, b) => a.organization.name.localeCompare(b.organization.name))
+
+  return (
+    <div>
+      {sortedTeams.map((team) => {
+        const {id: teamId, name: teamName, organization} = team
+        const {name: orgName} = organization
+
+        // TODO: implement filter on API side
+        const disabled = !!team.teamMembers.find((member) => member.userId === requestCreatedBy)
+        const active = selectedTeams.includes(teamId) || disabled
+
+        const handleClick = () => {
+          if (disabled) return
+
+          if (active) {
+            setSelectedTeams((prevSelectedTeams) => prevSelectedTeams.filter((id) => id !== teamId))
+          } else {
+            setSelectedTeams((prevSelectedTeams) => [...prevSelectedTeams, teamId])
+          }
+        }
+
+        return (
+          <div className='mb-2 flex items-center text-base' key={teamId}>
+            <Checkbox
+              active={active}
+              disabled={disabled}
+              onClick={handleClick}
+              color={active && !disabled ? PALETTE.SKY_500 : undefined}
+            />
+            <label className={`ml-2 ${disabled ? 'opacity-[.38]' : ''}`}>
+              {teamName} | {orgName}
```

Comment:
+1 would each label include the same `orgName`? If so, I think we should remove it here and include it in the subtitle, so: `Which teams in ${orgName} would you like to...`

![Screenshot 2023-04-25 at 16 27 20](https://user-images.githubusercontent.com/39854876/234353477-3d3749f7-d36d-41c9-81a6-311ff5b06d47.png)

Reply:
Not each label will include org name, because you may have multiple different orgs on the same domain
However, I agree that his could be improved. Added to the list of ideas for v2 if it is ok: https://www.notion.so/parabol/Do-you-want-to-join-x-org-prompt-V2-d09a562280b14b81bad54b87de31d3b0